### DISCUSSION TITLE
What is the best way to provide a count of unique items for a given day from millions of device accesses.

### DISCUSSION THREAD

**Discussion by bobskigit** [ravendb/ravendb#15098] [Q&A]
We need to provide a count of the unique devices that access the platform on a daily basis. There are millions of devices. Each device has a unique identifier that changes on a daily basis

Is it better to

1. Map index on date and then just linq distrinct in the backend API to provide a count of the unique devices.
2. Create a map-reduce unique the reduces by date and unique deviceId
3. Other ideas?

We are concerned that we create indexes that contain millions and millions of items.

Many thanks in advance

**Comment by ayende** [comment]
Map/Reduce my date & device id is easiest and most performing. 

What do you mean that the unique identifier changes on a daily basis, though?

**Reply by bobskigit** [reply]
We have a large advertising platform and the unique identifier changes on a daily basis to stop tracking. So each day the set of unique identifiers changes. My client whats to know how many unique devices we get on a given day.

**Reply by ayende** [reply]
Oh, I see. 
The way to do that is two folds. First, map/reduce on the date / id, then you do a faceted query to get the count of distinct ids per day.
